device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -432,10 +446,15 @@ public void registerConnectionStateCallback(IotHubConnectionStateCallback callba
         this.transport.registerConnectionStateCallback(callback, callbackContext);
     }
 
-    public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback statusChangeCallback, Object callbackContext)
+    public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback statusChangeCallback, Object callbackContext, String deviceId)
     {
         //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
-        this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
+        this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext, deviceId);
+    }
+
+    public void registerMultiplexingConnectionStateCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext)
```

Comment:
In which scenario would a user set one with a deviceId and without, or do both of them behave the same way? I see that this is publicly exposed under `DeviceIO`, but I don't think users access this directly, correct?

Reply:
This actually doesn't need to be public, so I'll make this package private

Reply:
The deviceId is specified to indicate that it will be called when connection status events occur to that specific device. For instance if the SAS token expires for just that device, then the user's device level callback would be fired

Reply:
```registerMultiplexingConnectionStateCallback``` on the other hand is strictly for multiplexed connection level events, like TCP connection drops